pynitrokey/cli/fido2.py

```diff
@@ -191,6 +191,114 @@ def list():
         local_print(f"{id_}: {name}")
 
 
+@click.command()
+@click.option(
+    "-s",
+    "--serial",
+    help="Serial number of Nitrokey to use. Prefix with 'device=' to provide device file, e.g. 'device=/dev/hidraw5'.",
+)
+@click.option("--pin", help="provide PIN instead of asking the user", default=None)
+def list_credentials(serial, pin):
+    """List all credentials saved on the key as well as the amount of remaining slots."""
+
+    # Gets device and client_pin object
+    device = nkfido2.find(serial)
+    client_pin = ClientPin(device.ctap2)
+
+    # Makes sure pin exists
+    if not pin:
+        pin = AskUser.hidden("Please provide pin: ")
+
+    try:
+        client_token = client_pin.get_pin_token(pin)
+    except Exception as e:
+        if "PIN_NOT_SET" in str(e):
```

Comment:
Can we use the error code directly, something like this?

```
except CtapError as e:
    if e.code == CtapError.ERR.PIN_NOT_SET:
```

Reply:
`if str(CtapError.ERR.PIN_NOT_SET) in str(e)` is the closest working solution I found

Reply:
Why doesn’t the direct comparison work?  I’ve tested it [here](https://github.com/Nitrokey/pynitrokey/commit/b44c195ca87653e70afcc98f560650965e2383b7) and it seems to work fine.

Reply:
Hmm, seems it slipped by me in testing, just changed it.